bin/ilm

```diff
@@ -147,25 +154,78 @@ ilm "(([a-g]\[aeiou]):0.1)^2"   "{256}.(2)"  # any sound space can be powered
     print('# '+' '.join(myargv))
     print('#')
 
-    for trial in range(options.num_trials):
-        parent = ilmpy.learners.AssociationMatrixLearner(*program_args,**program_kwargs)
-        if trial == 0:
-            parent.print_parameters()
-            if options.seed is not None:
-                print('# seed: {}'.format(options.seed))                
-            if options.amplitude is not None:
-                print('# amplitude: {}'.format(options.amplitude))
-            print('# bottleneck: {}\n# iterations: {}\n# trials: {}'.format(options.num_interactions,options.num_generations,options.num_trials))
-            print('# ')
-            parent.print_observables_header()
-        for generation in range(options.num_generations):
-            print('# Trial {} Iteration {}'.format(trial,generation))
-            child = parent.spawn()
-            #print("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXx")
-            lessons = parent.teach(options.num_interactions)
-            child.learn(lessons)
-            child.print_observables()
-            parent = child
-        
+
+    num_trials=options.num_trials
+    def multidef (num_trials,options):
+        result_dict={}
+        for trial in range(num_trials):
+            parent = ilmpy.learners.AssociationMatrixLearner(*program_args,**program_kwargs)
+            result_dict[trial]={}
+            print("Iter {}".format(trial), file=sys.stderr)
+            for generation in range(options.num_generations):
+#                if trial == 0:
+#                    parent.print_parameters()
+#                    if options.seed is not None:
+#                        print('# seed: {}'.format(options.seed))
+#                    if options.amplitude is not None:
+#                        print('# amplitude: {}'.format(options.amplitude))
+#                    print('# bottleneck: {}\n# iterations: {}\n# trials: {}'.format(options.num_interactions,options.num_generations,num_trials))
+#                    print('# ')
+#                    parent.print_observables_header()
+                for generation in range(options.num_generations):
+#                    print('# Trial {} Iteration {}'.format(trial,generation))
+                    child = parent.spawn()
+                    lessons = parent.teach(options.num_interactions)
+                    child.learn(lessons)
+                    result_dict[trial][generation]=child.print_observables()
+                    parent = child
+        return(result_dict,parent)
+
+
+    result_dict,parent=multidef (num_trials,options)
```

Comment:
If I remove line 185 I will get the following error:
    result=pool.starmap(result_dict,iter_jobs)         #(multidef,iter_jobs)
NameError: name 'result_dict' is not defined